account_invoice_export/models/account_move.py

```diff
@@ -0,0 +1,106 @@
+            return "Invoice has already been sent through http before."
+        if not self.transmit_method_id.send_through_http:
+            raise UserError(_("Transmit method is not configured to send through HTTP"))
+        file_data = self._get_file_for_transmission_method()
+        headers = self.transmit_method_id.get_transmission_http_header()
+        res = requests.post(
+            self.transmit_method_id.destination_url, headers=headers, files=file_data
+        )
+        if res.status_code != 200:
+            values = {
+                "job_id": self.env.context.get("job_uuid"),
+                "send_error": res.status_code,
+                "transmit_method_name": self.transmit_method_id.name,
+            }
+            with odoo.api.Environment.manage():
+                with odoo.registry(self.env.cr.dbname).cursor() as new_cr:
+                    # Create a new environment with new cursor database
+                    new_env = odoo.api.Environment(
+                        new_cr, self.env.uid, self.env.context
+                    )
+                    # The chatter of the invoice need to be updated, when the job fails
+                    self.with_env(new_env).log_error_sending_invoice(values)
+            raise UserError(
+                _(
+                    "HTTP error {} sending invoice to {}".format(
+                        res.status_code, self.transmit_method_id.name
+                    )
+                )
+            )
+        self.invoice_sent_through_http = True
+        self.invoice_send = True
+        self.log_success_sending_invoice()
+        return res.text
+
+    def _get_file_for_transmission_method(self):
+        """Return the file description to send.
+
+        Use the format expected by the request library
+        By default returns the PDF report.
+        """
+        r = self.env["ir.actions.report"]._get_report_from_name(
+            "account.report_invoice"
+        )
+        pdf, _ = r.render([self.id])
+        filename = self._get_report_base_filename().replace("/", "_") + ".pdf"
+        return {"file": (filename, pdf, "application/pdf")}
+
+    def log_error_sending_invoice(self, values):
+        """Log an exception in invoice's chatter when sending fails.
+
+        If an exception already exists it is update otherwise a new one
+        is created.
+
+        """
+        activity_type = "account_invoice_export.mail_activity_transmit_warning"
+        activity = self.activity_reschedule(
+            [activity_type], date_deadline=fields.Date.today()
+        )
+        if not activity:
+            message = self.env.ref(
+                "account_invoice_export.exception_sending_invoice"
+            ).render(values=values)
+            self.activity_schedule(
+                activity_type, summary="Job error sending invoice", note=message
+            )
+        else:
+            activity.note += "<p>It failed again, ouch!</p>"
```

Comment:
```suggestion
            activity.note += _("<p>It failed again, ouch!</p>")
```

Reply:
I fixed a little the messaging, but added the translation, thank you.